### PR TITLE
feat: add base swap page

### LINKED ISSUE #3553: Add swap interface
For users to swap between:

- BTC <-> x/sBTC
- STX <-> Stacks FTs
- Stacks FTs <-> Stacks FTs
- Lightning BTC <-> STX

Early mockup:

image

### PR DESCRIPTION
**Pull request by ldiego08**
## Overview

Add base page for token swap feature. This functionality is behind the `SWAP_ENABLED` feature flag.

Screenshot 2023-07-26 at 08 05 54

Screenshot 2023-07-26 at 08 06 04

> Try out this version of the Hiro Wallet - download [extension builds](https://github.com/hirosystems/wallet/actions/runs/5672136480).

### CODE REVIEW

File: src/app/features/swap/use-swap-feature.ts [hunk 1]
```diff
@@ -0,0 +1,5 @@
+export const useSwapFeature = () => {
+  return {
+    swapIsEnabled: !!process.env.SWAP_ENABLED
+  };
+}
```
Comment (kyranjamie) [inline src/app/features/swap/use-swap-feature.ts]: Please favour function declarations

```suggestion
export function useSwapFeature() {
  return {
    swapIsEnabled: !!process.env.SWAP_ENABLED
  };
}
```

**Comment by markmhendrickson** [comment]
Thanks @ldiego08! As you wire up screens on this PR, it'd be great to see screenshots in comments along the way (where relevant). 🙏

**Review by kyranjamie — COMMENTED** [review]
Sorry for slow review. General base set up looks good.

Re linting errors: some exports are unused, you may need to use `// ts-unused-exports:disable-next-line`, likely need prettier extension for editor as there are formatting violations, see errors for commit msgs, you'll need to rebase/reword these.

**Comment by fbwoolf** [comment]
@ldiego08 does it work for you if I just take this over and get it merged, then I can take over the UI from here?

**Comment by ldiego08** [comment]
@fbwoolf works perfect for me! Thank you!

**Comment by fbwoolf** [comment]
Closing this bc I am going to include the work from here in a new PR with more UI for swaps.

**Pull request closed without merging**